### PR TITLE
add 2 vm properties to allow storage persistence

### PR DESCRIPTION
**Pull request by jdoucerain**
in the VM spec
  image_pvc: name of an existing pvc to persist changes done on the first virtual drive (mounted on /image)
  volumes_pvc: name of an existing pvc to persist changes done on additional virtual drives (mounted on /volumes)

Signed-off-by: jdoucerain <jerome.doucerain@bell.ca>

### CODE REVIEW

**Comment by jdoucerain** [comment]
here is a sample manifest for a vm having persistent storage for both the image and additional volumes

apiVersion: vm.rancher.io/v1alpha1
kind: VirtualMachine
metadata:
  annotations:
  name: testvm2
spec:
  action: start
  cpus: 2
  hosted_novnc: true
  image: rancher/vm-rancheros:0.3.0
  image_pvc: image
  image_vmtools: ""
  kvm_extra_args: ""
  memory_mb: 4096
  node_name: ""
  public_keys:
  - jerome
  use_hugepages: false
  volumes_pvc: volumes

when deleting the VM and re-creating it, forcing it to be run on another node, I can get what was written to the virtual disk before deleting the VM

File: pkg/controller/vm/util.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ func (ctrl *VirtualMachineController) makeVMPod(vm *v1alpha1.VirtualMachine, ifa
 	var publicKeys []*v1alpha1.Credential
         var imagevmtools string
         var hugepagesvolume corev1.Volume
+        var vmimagevolume corev1.Volume
```
Comment (LLParse) [inline pkg/controller/vm/util.go]: Please use camelCase for your variables.

File: pkg/apis/ranchervm/v1alpha1/types.go [hunk 2]
```diff
@@ -51,7 +51,8 @@ type VirtualMachineSpec struct {
         KvmArgs string `json:"kvm_extra_args"`
         ImageVMTools string `json:"image_vmtools"`
         UseHugePages bool `json:"use_hugepages"`
-
+        VmImagePvc string `json:"image_pvc"`
```
Comment (LLParse) [inline pkg/apis/ranchervm/v1alpha1/types.go]: Field names are unclear, `VmImagePvcName` and `VmVolumesPvcName` would be better.

File: pkg/apis/ranchervm/v1alpha1/types.go [hunk 3]
```diff
@@ -51,7 +51,8 @@ type VirtualMachineSpec struct {
...
+        VmVolumesPvc string `json:"volumes_pvc"`
```
Comment (LLParse) [inline pkg/apis/ranchervm/v1alpha1/types.go]: Is Volumes plural here on purpose? I suppose if your plan is to use QCOW files as volumes, it doesn't matter if a single PV contains multiple QCOW volumes. We'll need to introduce new fields for block volume support anyways.

**Review by LLParse — CHANGES_REQUESTED** [review]
Looks good. Just a few minor tweaks.

File: pkg/apis/ranchervm/v1alpha1/types.go (hunk 3 above)
Comment (jdoucerain) [reply pkg/apis/ranchervm/v1alpha1/types.go]: The plural is there to indicate there could be multiple volumes in that single PVC

Comment (LLParse) [reply pkg/apis/ranchervm/v1alpha1/types.go]: I assumed that was the case. This is sufficient for a proof-of-concept, but performance won't be great. Consider your underlying PV has some IOPS guarantee which you would be effectively splitting between 1..N QCOW disk images.

That being said, QCOW likely won't perform as well as directly using a 3rd party storage provider...so let's not worry about performance now.

**Comment by jdoucerain** [comment]
variables names have changed

**Review by LLParse — APPROVED** [review]

**Comment by LLParse** [comment]
LGTM, thanks

**Pull request merged**